### DISCUSSION TITLE
Cluster node address use docker container name.

### DISCUSSION THREAD

**Discussion by montyclt** [ravendb/ravendb#14243] [Q&A] [answered]
```
System.IO.InvalidDataException: The database 'vcim' was created but is not accessible, because all of the nodes on which this database was supposed to reside on, threw an exception.
 ---> System.AggregateException: One or more errors occurred. (An exception occurred while contacting https://f6cea2ec78fc:8080/rachis/waitfor?index=10.
System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
 ---> Raven.Client.Exceptions.Security.CertificateNameMismatchException: You are trying to contact host f6cea2ec78fc but the hostname must match one of the CN or SAN properties of the server certificate: CN=vadavo.com, DNS:vadavo.com, DNS:vadavo.com
```

I installed RavenDB into a server through Docker, using my own SSL certificate that is valid for a domain name. When the server starts, it create a cluster with a single node and it using the container ID as an address.

My problem here is that the SSL certificate is not valid for the domain name that docker sets to the node. Can I configure the node address when initialising the container?

My application will not store a large amount of data, and a single node is sufficient.

**Comment by gregolsky** [comment, marked as answer]
It seems that you did not set the domain information in the RavenDB server settings. You can do that using `RAVEN_PublicServerUrl` environment variable or in *settings.json* file under `PublicServerUrl` key - the value has to match one of the SANs found in your certificate. Also don't forget to set `PublicServerUrl.Tcp` / `RAVEN_PublicServerUrl_Tcp` env var.

https://ravendb.net/docs/article-page/5.3/csharp/server/configuration/core-configuration#publicserverurl
https://ravendb.net/docs/article-page/5.3/csharp/server/configuration/core-configuration#publicserverurl.tcp

**Reply by montyclt** [reply]
Hi @gregolsky, thank you for your response. I set the PublicServerUrl and now I can create database from the studio.

Now, my application and the database services are defined in the same docker-compose file, and I want to connect it ideally using the default compose network, but certificate is only for the public URL.

Should I use public URL on the application or set insecure on internal network connections?

**NOTE**: The application and the database are deployed in a domain that is not in public DNS, only an internal DNS server can resolve it.

**Reply by gregolsky** [reply]
Your applications should use public URL to connect to RavenDB. 
If you don't want your DB connections to go outside and back through the public URL, you can try pointing your application container to use an internal address for your domain. I think you should be able to do that with `extra_hosts` docker-compose option: https://github.com/compose-spec/compose-spec/blob/master/spec.md#extra_hosts

**Reply by montyclt** [reply]
Thank you. I have one question more, about `PublicServerUrl_Tcp`. I'm not sure what value should I use here. I have only one node, so I don't need multiple nodes communicate between thems. The docker container is not exposing the TCP port. Should I provide a value?

**Reply by ayende** [reply]
The PublicServerUrl is there if you need a _different_ value than the ServerUrl.
For example, you you are binding to a PrivateIP and need to use the public ip DNS for replies.

In your scenario, I don't think this applies.
